### ISSUE TITLE
vector[double complex] causes compiler crash

### ISSUE THREAD

**Issue by isuruf** [cython/cython#1469]
This error occurs in v0.25.a0 but not in v0.24.1

Example file:
```cython
from libcpp.vector cimport vector
cdef vector[double complex] v
```

Error:
```cython

Error compiling Cython file:
------------------------------------------------------------
...

from libcpp.vector cimport vector
cdef vector[double complex] v
           ^
------------------------------------------------------------

test.pyx:3:12: Compiler crash in AnalyseDeclarationsTransform

File 'ModuleNode.py', line 118, in analyse_declarations: ModuleNode(test.pyx:2:0,
    full_module_name = 'test')
File 'Nodes.py', line 425, in analyse_declarations: StatListNode(test.pyx:2:0)
File 'Nodes.py', line 1290, in analyse_declarations: CVarDefNode(test.pyx:3:5,
    modifiers = [...]/0,
    visibility = 'private')
File 'Nodes.py', line 1123, in analyse: TemplatedTypeNode(test.pyx:3:12)

Compiler crash traceback from this point on:
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/Nodes.py", line 1123, in analyse
    self.type = base_type.specialize_here(self.pos, template_types)
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/PyrexTypes.py", line 3462, in specialize_here
    return self.specialize(dict(zip(self.templates, template_values)))
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/PyrexTypes.py", line 3480, in specialize
    specialized.scope = self.scope.specialize(values, specialized)
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/Symtab.py", line 2270, in specialize
    entry.type.specialize(values),
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/PyrexTypes.py", line 399, in specialize
    self.typedef_is_external, namespace)
  File "/home/isuru/.local/lib/python3.5/site-packages/Cython/Compiler/PyrexTypes.py", line 357, in __init__
    assert not base_type.is_complex
AssertionError:
```

**Comment by isuruf** [comment]
bisected this to 3f884a1cc9f9a706af5999327598e946ba7b081a

**Comment by robertwb** [comment]
Fixed by d6190df09c19032ec8c151ec7c79fce2fc05edf2.

**Comment by certik** [comment]
Thanks @robertwb for merging @isuruf's fix.
